src/subdomains/generic/kyc/services/name-check.service.ts

```diff
@@ -31,6 +35,44 @@ export class NameCheckService implements OnModuleInit {
     // void this.reloadSanctionList();
   }
 
+  @DfxCron(CronExpression.EVERY_MINUTE, { process: Process.NAME_CHECK_PDF_SYNC, timeout: 1800 })
+  async syncNameCheckPdf(): Promise<void> {
+    try {
+      const idsWithBankData = await this.nameCheckLogRepo
+        .createQueryBuilder('nameCheckLog')
+        .select('MIN(nameCheckLog.id)', 'minId')
+        .addSelect('nameCheckLog.bankDataId', 'bankDataId')
+        .addSelect('file.id', 'fileId')
+        .leftJoin('nameCheckLog.file', 'file')
+        .where('nameCheckLog.bankDataId IS NOT NULL')
+        .groupBy('nameCheckLog.bankDataId')
+        .addGroupBy('file.id')
+        .orderBy('file.id', 'DESC')
+        .getRawMany();
+
+      const ids = idsWithBankData.map((i) => i.minId);
+
+      const fullLogs = await this.nameCheckLogRepo.find({
+        where: { id: In(ids) },
+        relations: { bankData: true, file: true },
+      });
+    } catch (e) {
+      console.log(e);
+    }
+
+    const entities = await this.nameCheckLogRepo.find({
+      order: { file: { id: 'ASC' } },
+      take: 5000,
+      relations: { bankData: true },
+    });
+
+    for (const entity of entities) {
+      if (entity.file) continue;
```

Comment:
Need to move this if to the SQL query? What does prevent getting the same 5000 entries every time?